packages/app-extensions/src/rest/helpers/preferences.js

```diff
@@ -0,0 +1,31 @@
+import {call} from 'redux-saga/effects'
+
+import {requestSaga} from '../rest'
+
+/**
+ * To fetch user preferences.
+ *
+ * @param path {string} To load specific preferences matching the path. Wildcards (e.g. *) may be used.
+ */
+export function* fetchUserPreferences(path) {
+  const response = yield call(requestSaga, `client/preferences${path ? '/' + path : ''}`)
+  return response.body.preferences
+}
+
+/**
+ * Deletes all preferences matching the path.
+ *
+ * @param path {string} All preferences matching the path will be deleted. Wildcards (e.g. *) may be used.
+ */
+export function* deleteUserPreferences(path) {
+  yield call(requestSaga, `client/preferences/${path}`, {method: 'DELETE'})
+}
+
+/**
+ * To add or overwrite preferences.
+ *
+ * @param preferences {object} Key/Value pairs.
+ */
+export function* savePreferences(preferences) {
+  yield call(requestSaga, 'client/preferences', {method: 'PUT', body: {values: preferences}})
```

Comment:
this will update preferences only partially?

e.g. if there are the following preferences already saved:
- foo: "bar"
- dings: 5

and you send with your request:
- dings: 6
- newpref: "hello"

will the saved preferences after your request be {foo: "bar", dings: 6, newpref: "hello"} or will they only be {dings: 6, newpref: "hello"} (only the data you sent)?

if it's the first, the HTTP method PATCH should be used instead of PUT. if it's the latter, PUT is correct.

Reply:
ok we will change this to PATCH